src/naf-dialog-adapter.js

```diff
@@ -99,6 +102,62 @@ export default class DialogAdapter {
     this._onOccupantMessage = messageListener;
   }
 
+  async getServerStats() {
+    const result = {};
+    try {
+      if (this._sendTransport && !this._sendTransport._closed && this._sendTransport._connectionState == "connected") {
+        result[this._sendTransport.id] = {};
+        result[this._sendTransport.id]["name"] = "Send";
+        result[this._sendTransport.id]["stats"] = await this._protoo.request("getTransportStats", {
+          transportId: this._sendTransport.id
+        });
+        result[this._sendTransport.id]["producers"] = {};
+        for (const producer of this._sendTransport._producers) {
+          const id = producer[0];
+          result[this._sendTransport.id]["producers"][id] = await this._protoo.request("getProducerStats", {
+            producerId: id
+          });
+        }
+      }
+      if (this._recvTransport && !this._recvTransport._closed && this._recvTransport._connectionState == "connected") {
+        result[this._recvTransport.id] = {};
+        result[this._recvTransport.id]["name"] = "Receive";
+        result[this._recvTransport.id]["stats"] = await this._protoo.request("getTransportStats", {
+          transportId: this._recvTransport.id
+        });
+        result[this._recvTransport.id]["consumers"] = {};
+        for (const consumer of this._recvTransport._consumers) {
+          const id = consumer[0];
+          result[this._recvTransport.id]["consumers"][id] = await this._protoo.request("getConsumerStats", {
+            consumerId: id
+          });
+        }
+      }
+      return result;
+    } catch (e) {
+      this.scene.emit("rtc_event", {
+        level: "error",
+        tag: "Adapter",
+        msg: `Error getting the server status: ${e}`
+      });
+      return { error: `Error getting the server status: ${e}` };
+    }
+  }
+
+  async restartSendICE() {
+    if (this._sendTransport !== null && !this._sendTransport._closed) {
```

Comment:
comment

Reply:
This can also be simplified. Updated and comment added.